G1ANT.Addon.Selenium/Api/SeleniumWrapper.cs

```diff
@@ -364,6 +364,42 @@ public string GetAttributeValue(string attributeName, SeleniumCommandArguments s
             return res;
         }
 
+        public string GetHtml(SeleniumIFrameArguments search, TimeSpan timeout)
+        {
+            PreCheckCurrentWindowHandle();
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
+            var element = FindElement("//html", "xpath", timeout);
+            var res = element?.GetAttribute("outerHTML") ?? string.Empty;
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().DefaultContent();
+            return res;
+        }
+
+        public string GetInnerHtml(SeleniumCommandArguments search, TimeSpan timeout)
+        {
+            PreCheckCurrentWindowHandle();
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
+            var element = FindElement(search.Search.Value, search.By.Value, timeout);
+            var res = element?.GetAttribute("innerHTML") ?? string.Empty;
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().DefaultContent();
+            return res;
+        }
+
+        public string GetOuterHtml(SeleniumCommandArguments search, TimeSpan timeout)
+        {
+            PreCheckCurrentWindowHandle();
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
+            var element = FindElement(search.Search.Value, search.By.Value, timeout);
+            var res = element?.GetAttribute("outerHTML") ?? string.Empty;
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().DefaultContent();
+            return res;
+        }
+
```

Comment:
this methods looks like DRY violation, BTW you can use GetAttributeValue() method without creating new ones

Reply:
Jasne